### DISCUSSION TITLE
Addition of Reduced Height / Truncated Top of Display CFG Command for 0.96" 128x64 OLED

### DISCUSSION THREAD

**Discussion by SynthfulDwarphus** [keirf/flashfloppy#671] [Ideas]
Hello everyone,

I propose a FlashFloppy display CFG Command to reduce /remove / truncate the top few horizontal display lines of the 0.96" 128x64 OLED so that there are no unviewable displayed lines on the OLED when mounted in the factory standard Gotek housing cut-out window.

While the 0.96 OLED can indeed be made to fit its entire 64-pixel vertical height to be viewable in the factory window, doing so requires the difficulties and dangers fully flattening (via difficult sanding/cutting) of the Gotek housing floor's little raised lip at the inside corner where it meets the inside front panel, and even then, to sharply bend and compress the OLED ribbon cable tightly when mounting the OLED panel's bottom edge completely flush against the floor.

That done, and if the extremely bent ribbon hasn't yet destroyed the display, the very top line of the OLED will _just_ be visible if viewing the display completely head-on, made more difficult to see by the recessed nature of the mounting, and the reflection from the glossy plastic on the inside edges of the window.

However, this issue is avoidable if the top two, four, six, or eight horizontal graphic lines of FlashFloppy menu's output could be truncated. Since FlashFloppy only displays maximum four lines of text, leaving quite a bit of space between character lines no matter which font is selected, the two to eight lines can likely easily be made up with reduced empty space between those text lines.

A CFG display option command to accomplish this might be something like (for removing four lines) "-4HLines" or "-4height" or "top-4", etc.

ATTACHED TO THIS POST is a pic of a slightly crooked (it looks better in person) 0.96 OLED in standard factory Gotek window, where all top-most horizontal graphic lines are actually visible (maybe 1-2 lines gone at right side). 
![IMG_20220714_234129094~2](https://user-images.githubusercontent.com/105606636/180676261-740d420e-0f24-48ac-aa56-ea8f7b95160a.jpg)

An option to remove just four to eight top display lines would make mounting of a 0.96 OLED much easier and safer, and improve viability for many users. I'm not a programmer at all, so I would appreciate someone who is skilled implementing the new feature. 

Open to all comments, and thanks for considering!

**Comment by keirf** [comment]
Using the FF.CFG `display-order=` option you can simply leave the top row (16px) blank. There is no option, or point, in shaving off fewer pixels as FlashFloppy deals in text rows and that's multiples of 16px.

**Comment by SynthfulDwarphus** [comment]
Thanks for the amazing firmware, and all your continued support, Keir!

In my case (I'm sure mirrored by most users), the entire point of modding
with the .96 OLED is to have BOTH a much more visible, bigger,
double-height font (i.e. 8x16) in the small display space, and have a
variety of info regarding the disk/directory/status/etc. So, EVERY line of
text is critical (either a 2nd double-height 8x16 one, or two small 6x13
ones).

There's lots of empty space on the display as it is, and to remove a
further full 25% of it by leaving the top 16 pixels blank, as you suggest,
defeats a lot of the purpose and utility of the .96 OLED.

I think removal of 4 pixels (and possibly also a -8 option) would be all
that is necessary to both make mounting and seeing the OLED better.

With that in mind, is it at all possible to make FlashFloppy firmware also
have the ability to utilize four 15-pixel text lines (60 total lines, with
the remaining top 4 lines blank), and maybe four 14-pixel text lines (56
lines, with top 8 black)?

The text lines would still be sufficiently spaced, I think.

I'm not sure how hard this would be to implement, but I think anyone who
has, or will, mod with a .96 will then be able to utilize and comfortably
view essentially all of the display.

Thanks again for your consideration and time!

Best,
SynD
